model/host/host.go

```diff
@@ -304,6 +304,9 @@ type ProvisionOptions struct {
 	// artifacts for the given task.
 	TaskId string `bson:"task_id" json:"task_id"`
 
+	//FetchOps contains data needed to fetch task data
+	FetchOpts *FetchOpts `bson:"fetch_options,omitempty" json:"fetch_options,omitempty"`
```

Comment:
It would simplify things a lot (both for this PR and for the future) if we did the thing I proposed in my other comment where we 1. revoke in the CLI command itself rather than on the app server side, and then 2. rather than initializing the fetch options when the host is inserted into the DB, move all the logic to figure out the owner/repo/modules/tokens into [this function](https://github.com/evergreen-ci/evergreen/blob/37d5980bdca8f361cec36529303df6da98ae9323/model/host/hostutil.go#L1104-L1118). By doing that, we could lazily populate the tokens right when the `evergreen fetch` command is being made and pass the tokens along to the command. The command will eventually revoke them itself. It also has the advantage of simplicity, because instead of having to pass around all this host state all over the place, we only use it in the one place where it's actually relevant (i.e. making the `evergreen fetch` command).

Doing it that way also has a big advantage of avoiding persisting GitHub tokens in plaintext in the DB. Since we're trying to move away from storing plaintext secrets in the DB, storing the tokens like this in the host document will eventually become tech debt that we have to figure out how to move out of the DB later on.